### DISCUSSION TITLE
LDM Notes for May 8th, 2024

### DISCUSSION THREAD

**Discussion by 333fred** [dotnet/csharplang#8107] [LDM Notes]
https://github.com/dotnet/csharplang/blob/main/meetings/2024/LDM-2024-05-08.md

## Agenda

- `readonly` for primary constructor parameters

**Comment by alrz** [comment]
Is changing the default also an option here? Currently records are intentionally opinionated and for anything outside the default you'd need to write your own class. I think this can apply to primary constructors, if you want mutable fields, you could declare mutable fields.
```cs
class C(int readonlyByDefault) {
  private int mutable = readonlyByDefault; // plus, capturing readonlyByDefault in this case is already a warning
}

**Reply by CyrusNajmabadi** [reply]
Yes. That's something we are interested in investigating. Part of a larger push around immutability.

**Reply by KennethHoff** [reply]
Option for Immutability by default; you heard it here first!

**Reply by HaloFour** [reply]
> Yes. That's something we are interested in investigating. Part of a larger push around immutability.

Hmmm, your ideas are intriguing to me and I wish to subscribe to your newsletter.

**Reply by CyrusNajmabadi** [reply]
![image](https://github.com/dotnet/csharplang/assets/4564579/e2911026-6387-45bb-8355-550c95933e4d)

**Reply by TahirAhmadov** [reply]
In my code base, 100% (not exaggerating) of primary ctor parameters are "readonly" by actual usage. Personally, I fully support the default readonly approach.

**Comment by qrli** [comment]
I don't like the Analyzer solution. If that direction turned out to be the best choice, I'd at least like to have __compiler warnings__ for assignments to primary constructor parameters.

PS: Thanks for your effort to push this forward.

**Reply by CyrusNajmabadi** [reply]
Analyzers can produce compiler warnings.
